examples-dot-net/NewREADME.md

```diff
@@ -0,0 +1,428 @@
+
+## Documentation
+
+Documentation is available to help you learn how to use this package:
+
+- [Quickstart](https://github.com/Azure/azure-sdk-for-net/blob/main/doc/dev/mgmt_quickstart.md).
+- [API References](https://docs.microsoft.com/dotnet/api/?view=azure-dotnet).
+- [Authentication](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/README.md).
+
+## Examples
+
+### Extension Management  
+
+##### Prerequisites
+1. Get the Azure token
+
+```C# Snippet: 
+            TokenCredential cred = new DefaultAzureCredential();
+            ArmClient client = new ArmClient(cred);
+```
+2. Update the parameters given below
+```C# Snippet:
+            string subscription = "00000000-0000-0000-0000-000000000000"; # Replace with your subscription ID
+            string resourceGroupName = "hcicluster-rg"; # Replace with your resource group name
+            string clusterName = "HCICluster"; # Replace with your cluster name
+            string arcSettingName = "HCIArcSettingName"; # Replace with your Arc Setting Name
+            string extensionName = "MicrosoftMonitoringAgent"; # Replace with your extension name
+```
+3. Get Arc Extension Resource
+
+```C# Snippet: 
+            ResourceIdentifier arcExtensionResourceId = ArcExtensionResource.CreateResourceIdentifier(subscriptionId, resourceGroupName, clusterName, arcSettingName, extensionName);
+            ArcExtensionResource arcExtension = client.GetArcExtensionResource(arcExtensionResourceId);
+```
+
+#### Installing Extensions as part of enabling capabilities
+
+##### Prerequisites
+
+1. Get the Azure token
+
+```C# Snippet: 
+            TokenCredential cred = new DefaultAzureCredential();
+            ArmClient client = new ArmClient(cred);
+```
+2. Update the parameters given below
+```C# Snippet:
+            string subscription = "00000000-0000-0000-0000-000000000000"; # Replace with your subscription ID
+            string resourceGroupName = "hcicluster-rg"; # Replace with your resource group name
+            string clusterName = "HCICluster"; # Replace with your cluster name
+            string arcSettingName = "HCIArcSettingName"; # Replace with your Arc Setting Name
+```
+3. Get Arc Setting Resource
+
+```C# Snippet: 
+            ResourceIdentifier arcSettingResourceId = ArcSettingResource.CreateResourceIdentifier(subscriptionId, resourceGroupName, clusterName, arcSettingName);
+            ArcSettingResource arcSetting = client.GetArcSettingResource(arcSettingResourceId);
+```
+
+##### Install AMA
+
+1. Create the Payload
+
+```C# Snippet:
+            // invoke the operation
+
+            string extensionName = "AzureMonitorWindowsAgent";
+            ArcExtensionData data = new ArcExtensionData()
+            {
+                Publisher = "Microsoft.Azure.Monitor",
+                ArcExtensionType = "AzureMonitorWindowsAgent",
+                TypeHandlerVersion = "1.10",
```

Comment:
make all of the parameters that you are passing in the payload variables and define them above